x-pack/platform/test/alerting_api_integration/spaces_only/tests/helpers.ts

```diff
@@ -25,3 +28,99 @@ export async function tearDown(getService: FtrProviderContext['getService']) {
   const spacesService = getService('spaces');
   for (const space of Object.values(Spaces)) await spacesService.delete(space.id);
 }
+
+export const runSoon = async ({
+  id,
+  supertest,
+  retry,
+  spaceId = Spaces.space1.id,
+}: {
+  id: string;
+  supertest: SuperTestAgent;
+  retry: RetryService;
+  spaceId?: string;
+}) => {
+  return retry.try(async () => {
+    await supertest
+      .post(`${getUrlPrefix(spaceId)}/internal/alerting/rule/${id}/_run_soon`)
+      .set('kbn-xsrf', 'foo')
+      .expect(204);
+  });
+};
+
+export const getRuleEvents = async ({
+  id,
+  action,
+  newInstance,
+  activeInstance,
+  recoveredInstance,
+  retry,
+  spaceId = Spaces.space1.id,
+  getService,
+}: {
+  id: string;
+  action?: number;
+  newInstance?: number;
+  activeInstance?: number;
+  recoveredInstance?: number;
+  retry: RetryService;
+  spaceId?: string;
+  getService: FtrProviderContext['getService'];
+}) => {
+  const actions: Array<[string, { equal: number }]> = [];
+  if (action) {
+    actions.push(['execute-action', { equal: action }]);
+  }
+  if (newInstance) {
+    actions.push(['new-instance', { equal: newInstance }]);
+  }
+  if (activeInstance) {
+    actions.push(['active-instance', { equal: activeInstance }]);
+  }
+  if (recoveredInstance) {
+    actions.push(['recovered-instance', { equal: recoveredInstance }]);
+  }
+  return retry.try(async () => {
+    return await getEventLog({
+      getService,
+      spaceId,
+      type: 'alert',
+      id,
+      provider: 'alerting',
+      actions: new Map(actions),
+    });
+  });
+};
+
+export const waitForRuleExecute = ({
```

Comment:
nit: I'm surprised we didn't already have a function for this, but didn't see one.  Since it also takes action executions into account, perhaps adding that in the name?  `waitForRuleExecuteWithActions`?